src/ucp/wireup/wireup_cm.c

```diff
@@ -279,7 +280,15 @@ static unsigned ucp_cm_client_connect_progress(void *arg)
        since TL can be changed due to server side configuration */
     tl_bitmap = ucp_ep_get_tl_bitmap(ucp_ep);
     ucs_assert(tl_bitmap != 0);
-    dev_index = context->tl_rscs[ucs_ffs64(tl_bitmap)].dev_index;
+    rsc_index = ucs_ffs64(tl_bitmap);
+    dev_index = context->tl_rscs[rsc_index].dev_index;
+
+#if ENABLE_ASSERT
```

Comment:
is this `#if` really needed?

Reply:
`ucs_for_each_bit` has asm code, so I don't think compiler is able to compile out such empty loop